tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -171,28 +191,88 @@ def __validate_section2(self, num_errors):
         num_errors += self.__validate_t5_atd_and_ssi()
         return num_errors
 
-    def __validate_family_affiliation(self, num_errors, t1s, t2s, t3s, error_msg):
+    def __validate_family_affiliation(self,
+                                      num_errors,
+                                      t1_model_name, t1s,
+                                      t2_model_name, t2s,
+                                      t3_model_name, t3s):
         """Validate at least one record in t2s+t3s has FAMILY_AFFILIATION == 1."""
         num_errors = 0
         passed = False
-        for record, schema in t2s + t3s:
-            family_affiliation = getattr(record, 'FAMILY_AFFILIATION')
+        error_msg = (
+                        f"Every {t1_model_name} record should have at least one corresponding "
+                        f"{t2_model_name} or {t3_model_name} record with the same "
+                    )
+
+        is_t2 = True
+        t2_context = self.__get_error_context("FAMILY_AFFILIATION", t2s[0][1]) + "==1"
+        for record, schema, line_num in t2s:
+            family_affiliation = getattr(record, "FAMILY_AFFILIATION")
             if family_affiliation == 1:
                 passed = True
+                is_t2 = False
                 break
 
+        is_t3 = True
+        t3_context = self.__get_error_context("FAMILY_AFFILIATION", t3s[0][1]) + "==1"
+        for record, schema, line_num in t3s:
+            family_affiliation = getattr(record, "FAMILY_AFFILIATION")
+            if family_affiliation == 1:
+                passed = True
+                is_t3 = False
+                break
+
+        final_context = ""
+        if is_t2 and is_t3:
+            final_context += t2_context + " and " + t3_context + "."
```

Comment:
@elipe17 is this the line of code that constructs the message below? If so, can you change "and" to "or"? basically, at least one of the person-level records has to have affil = 1, not necessarily both. 

`Every T1 record should have at least one corresponding T2 or T3 record with the same Item 4 (Reporting Year and Month) and Item 6 (Case Number), where Item 30 (Family Affiliation)==1 and Item 67 (Family Affiliation)==1.`